tests/test_algebra_cascade.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+from distutils.version import StrictVersion
+import numpy as np
+from numpy.testing import assert_almost_equal
+import onnx
+from onnxruntime import InferenceSession
+from skl2onnx.common.data_types import FloatTensorType
+from skl2onnx.algebra.onnx_ops import OnnxAdd, OnnxScaler
+
+
+class TestOnnxOperatorsCascade(unittest.TestCase):
+
+    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion("1.4.0"),
+                     reason="not available")
+    def test_cascade_add(self):
+
+        def generate_onnx_graph(dim, nbnode, input_name='X1'):
+            i1 = input_name
+            for i in range(nbnode - 1):
+                i2 = (np.ones((1, dim)) * nbnode * 10).astype(np.float32)
+                node = OnnxAdd(i1, i2)
+                i1 = node
+            i2 = (np.ones((1, dim)) * nbnode * 10).astype(np.float32)
+            node = OnnxAdd(i1, i2, output_names=['Y'])
+            onx = node.to_onnx([(input_name, FloatTensorType((1, dim)))],
+                               outputs=[('Y', FloatTensorType())])
+            return onx
+
+        exp = [np.array([[11., 11., 11., 11., 11.]]),
```

Comment:
Why don't you use np.full instead?

Reply:
Easier to read this way. It is a unit test.